src/main/java/duchess/Duchess.java

```diff
@@ -0,0 +1,229 @@
+package duchess;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File; 
+import java.io.BufferedWriter; 
+import java.io.BufferedReader; 
+import java.io.FileWriter; 
+import java.io.FileReader; 
+
+public class Duchess {
+    boolean isRunning;
+    private static final String FILE_PATH = "./data/list.txt";
+
+
+    TaskList taskList;
+    String in;
+    Ui ui;
+    Parser parser;
+    Storage storage;
+
+    /**
+     * Constructs a new Duchess application instance.
+     * Initializes UI, parser, storage, and loads the task list.
+     */
+    public Duchess() {
+        this.ui = new Ui();
+        this.parser = new Parser();
+        this.isRunning = true;
+        this.storage = new Storage(FILE_PATH);
+        this.taskList = this.storage.loadList();
+    }
+
+    /**
+     * Adds a Todo task to the task list.
+     *
+     * @param in The user input containing the task description.
+     * @throws DuchessException If the input format is invalid.
+     */
+    public void addTodo(String in) throws DuchessException  {
```

Comment:
There is an extra space here.
```suggestion
    public void addTodo(String in) throws DuchessException {
```